core/shared/src/main/scala/spire/math/poly/MultivariatePolynomial.scala

```diff
@@ -0,0 +1,214 @@
+package spire.math.poly
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuilder
+import scala.collection.mutable.Set
+import scala.reflect._
+import scala.{specialized => spec}
+import spire.algebra._
+import spire.implicits._
+import spire.math._
+
+
+class MultivariatePolynomial[@spec(Double) C: Order] private[spire] (val terms: Array[Monomial[C]])
```

Comment:
I'll try to get rid of `Order`.  Also, pardon my ignorance, I had thought that making the constructor private also made the class members in the constructor private.  I'll go through and make `terms` private here and do the same thing for Monomial's class members and make sure there are methods to access them.  These should be hidden implementation details.